### PR TITLE
feat: kubernetes connection store

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

The connection status of the current context must be accessible to the frontend. This PR is adding a `kubernetesConnection` store.

### What issues does this PR fix or reference?

Related to https://github.com/containers/podman-desktop/issues/5401

### How to test this PR?

- [x] Unit tests has been added

### CODE REVIEW

File: packages/main/src/plugin/kubernetes-connection.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
```
Comment (deboer-tim) [inline packages/main/src/plugin/kubernetes-connection.ts:2]: Nit: 2024

**Review by deboer-tim — CHANGES_REQUESTED** [review]
+1 to this function, we really need it.

My main concern is that to get the list of contexts or what the current one is you use stores/kubernetes-contexts, but then to get the status of the current context you use another store that also says what the context is. When changing current context it's not clear how in sync these are or whether/how clients would deal with when the connection context isn't the current context.

There's also PR #5329 which overlaps on connection status. The one nice aspect there is that it's a map, so it doesn't duplicate what the current context is and can store connection info for multiple contexts.

Maybe a group chat would help, or you could discuss with @feloy?

File: packages/renderer/src/stores/kubernetes-connection.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { writable, type Writable } from 'svelte/store';
+import { EventStore } from './event-store';
+import { checkForUpdate } from './kubernetes-contexts';
+import type { KubernetesConnection } from '../../../main/src/plugin/kubernetes-connection';
+
+const windowEvents: string[] = ['kubernetes-context-update', 'kubernetes-connection-update'];
```
Comment (jeffmaury) [inline packages/renderer/src/stores/kubernetes-connection.ts:24]: I don't see where kubernetes-connection-update is being sent

File: packages/main/src/plugin/index.ts [hunk 3]
```diff
@@ -2009,6 +2010,12 @@ export class PluginSystem {
       return kubernetesClient.getDetailedContexts();
     });
 
+    this.ipcHandle('kubernetes-client:getConnectionStatus', async (): Promise<KubernetesConnection | undefined> => {
+      const value = kubernetesClient.getConnectionStatus();
+      console.log('getConnectionStatus', value);
```
Comment (jeffmaury) [inline packages/main/src/plugin/index.ts:2015]: To be removed

**Comment by axel7083** [comment]
Closing in favour of https://github.com/containers/podman-desktop/pull/5917/files#diff-ea47798d4fd89a4b135b3c7c7e1be18d4ba792cb8dd900425946693f0e7a53c9

**Pull request closed without merging**